torch/distributions/distribution.py

```diff
@@ -121,6 +121,13 @@ def mean(self):
         """
         raise NotImplementedError
 
+    @property
+    def mode(self):
+        """
+        Returns the mode of the distribution.
+        """
+        raise NotImplementedError(self)
```

Comment:
nit: why not simply `raise NotImplementedError`?

Reply:
Sorry, this was left over from development so that it was obvious which distribution was causing the tests to fail. Will remove in an update.

Reply:
If this is intended to give a more specific error message, how about `NotImplementedError(f'{self.__class__} does not define mode')`?